### PR TITLE
Add slider component

### PR DESCRIPTION
**Pull request by Fires1**

### CODE REVIEW

File: code/ui/components/slider/Slider.cs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+namespace TTTReborn.UI
+{
+    /// <summary>
+    /// A horizontal slider. Can be float or whole number.
+    /// </summary>
+    public class Slider : TTTPanel
+    {
+        public Panel Track { get; protected set; }
+        public Panel TrackInner { get; protected set; }
+        public Panel Thumb { get; protected set; }
```
Comment (mzegar) [inline code/ui/components/slider/Slider.cs:17]: Should these be public? What's the use case for accessing them outside. Changing the style?

File: code/ui/components/slider/Slider.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        public float MinValue { get; set; } = 0;
+
+        /// <summary>
+        /// If set to 1, value will be rounded to 1's
+        /// If set to 10, value will be rounded to 10's
+        /// If set to 0.1, value will be rounded to 0.1's
+        /// </summary>
+        public float Step { get; set; } = 1.0f;
+
+        public Slider()
```
Comment (mzegar) [inline code/ui/components/slider/Slider.cs:36]: Do we want to make this private/protected? Would we want users to be able to use this constructor?

**Review by mzegar — COMMENTED** [review]
Also a few `var` that will need to be converted.

File: code/ui/components/slider/Slider.cs (hunk 1 above)
Comment (Fires1) [reply code/ui/components/slider/Slider.cs:17]: Retrieving data off of them.

File: code/ui/components/slider/Slider.cs (hunk 2 above)
Comment (Fires1) [reply code/ui/components/slider/Slider.cs:36]: I say leave it.

**Review by mzegar — APPROVED** [review]

**Pull request merged**